libs/@blockprotocol/graph/src/stdlib/bound.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  NonNullTimeInterval,
+  TemporalBound,
+} from "../types/temporal-versioning.js";
+
+export const compareBounds = (
+  lhs: TemporalBound,
+  rhs: TemporalBound,
+  lhsType: keyof NonNullTimeInterval,
+  rhsType: keyof NonNullTimeInterval,
+): number => {
+  // If the bound values are not equal, then the bound with the lower value is less than the bound with the higher
+  // value.
+  if (lhs.kind !== "unbounded" && rhs.kind !== "unbounded") {
+    if (lhs.limit !== rhs.limit) {
+      return lhs.limit.localeCompare(rhs.limit);
+    }
+  }
+
+  if (lhs.kind === rhs.kind && lhsType === rhsType) {
+    return 0;
+  }
+
+  if (
+    (lhs.kind === "unbounded" && lhsType === "start") ||
+    (rhs.kind === "unbounded" && rhsType === "end") ||
+    (lhs.kind === "exclusive" &&
+      rhs.kind === "exclusive" &&
+      lhsType === "end" &&
+      rhsType === "start") ||
+    (lhs.kind === "exclusive" &&
+      rhs.kind === "inclusive" &&
+      lhsType === "end") ||
+    (lhs.kind === "inclusive" &&
+      rhs.kind === "exclusive" &&
+      rhsType === "start")
+  ) {
+    return -1;
+  }
+
+  if (
+    (lhs.kind === "unbounded" && lhsType === "end") ||
+    (rhs.kind === "unbounded" && rhsType === "start") ||
+    (lhs.kind === "exclusive" &&
+      rhs.kind === "exclusive" &&
+      lhsType === "start" &&
+      rhsType === "end") ||
+    (lhs.kind === "exclusive" &&
+      rhs.kind === "inclusive" &&
+      lhsType === "start") ||
+    (lhs.kind === "inclusive" && rhs.kind === "exclusive" && rhsType === "end")
+  ) {
+    return 1;
+  }
+  throw new Error(
+    `Implementation error, failed to compare bounds.\nLHS: ${JSON.stringify(
+      lhs,
+    )}\nRHS: ${JSON.stringify(rhs)}`,
+  );
+};
+
+/**
+ * Checks whether two given temporal bounds are adjacent to one another, where adjacency is defined as being next to one
+ * another on the timeline, without any points between, *and where they are not overlapping*.
+ *
+ * As such, two inclusive (or two exclusive) bounds with the same limit are *not* adjacent.
+ *
+ * @param lhs - The first bound of the comparison (order is unimportant)
+ * @param rhs - The second bound of the comparison
+ */
+export const boundIsAdjacentTo = (
```

Comment:
The function is not called on a `bound` but is a free standing function. Maybe we should rename it to `boundsAreAdjacent`?

Reply:
You are correct that it is not a method, but that is because we aren't using classes. I'd prefer the naming convention I've used here because for quite a few of them the ordering of the parameters matters, I admit for a few of these it does not (this for example), but for some of them like checking if an interval contains an interval.

Reply:
I also prefer `areBoundsAdjacent` (putting `are` first because it's more consistent with boolean fields that are called `isSomething`).

I would expect `boundIsAdjacentTo` to return the thing that a bound is adjacent to. So at minimum I think this could be called `boundIsAdjacentToBound`

Reply:
> So at minimum I think this could be called boundIsAdjacentToBound

Reply:
I personally think starting the function with the object makes it easier to find what you're looking for

Reply:
Fine with either

If you go with `areXY` note that the first entry in the table should be `areBoundsAdjacent`

Reply:
I've renamed `boundIsAdjacentTo` to `boundIsAdjacentToBound`.

I'm fine going with the other side of the table if someone feels stronger about this than me, but for now I'm leaning to leaving it as is